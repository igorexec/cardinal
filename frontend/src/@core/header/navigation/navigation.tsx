import React from 'react';
import {NavLink} from 'react-router-dom';
import {NavigationItem} from '../types';

interface Props {
  items: NavigationItem[];
}

export const Navigation: React.FC<Props> = ({items}) => {
  const navItems = items.map(({id, link, title}) => <NavLink key={id} to={link}>{title}</NavLink>);

  return (
    <nav>
      <ul>{navItems}</ul>
    </nav>
  );
};
