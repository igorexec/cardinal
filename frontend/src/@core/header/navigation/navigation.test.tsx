import React from 'react';
import {shallow, ShallowWrapper} from 'enzyme';
import {Navigation} from './navigation';

describe('Navigation', () => {
  let wrapper: ShallowWrapper;
  const props = {items: [{id: '1', title: 'title', link: 'link'}]};

  beforeAll(() => {
    wrapper = shallow(<Navigation {...props} />);
  });

  it('matches snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
