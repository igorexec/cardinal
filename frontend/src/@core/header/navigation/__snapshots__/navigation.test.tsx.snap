// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation matches snapshot 1`] = `
<nav>
  <ul>
    <NavLink
      key="1"
      to="link"
    >
      title
    </NavLink>
  </ul>
</nav>
`;
